Source/SuperLinq/Exclude.cs

```diff
@@ -46,50 +46,158 @@ public static IEnumerable<T> Exclude<T>(this IEnumerable<T> sequence, int startI
+	/// </remarks>
+	public static IEnumerable<T> Exclude<T>(this IEnumerable<T> sequence, Range range)
+	{
+		ArgumentNullException.ThrowIfNull(sequence);
+		ArgumentOutOfRangeException.ThrowIfNegative(range.Start.Value);
+		ArgumentOutOfRangeException.ThrowIfNegative(range.End.Value);
+
+		if (range.Start.Value - range.End.Value == 0)
+			return sequence;
+
+		return sequence switch
+		{
+			IList<T> list => new ExcludeListIterator<T>(list, range),
+			ICollection<T> collection => new ExcludeCollectionIterator<T>(collection, range),
+			_ => ExcludeCore(sequence, range)
+		};
+	}
+
+	/// <summary>
+	/// Represents an iterator for excluding elements from a collection.
+	/// </summary>
+	/// <typeparam name="T">Specifies the type of elements in the collection.</typeparam>
+	private sealed class ExcludeCollectionIterator<T>
+		: CollectionIterator<T>
 	{
-		public override int Count =>
-			source.Count < startIndex ? source.Count :
-			source.Count < startIndex + count ? startIndex :
-			source.Count - count;
+		private readonly ICollection<T> _source;
+		private readonly int _startIndex;
+		private readonly int _count;
 
+		/// <summary>
+		/// Creates a new collection iterator for excluding elements from the specified <paramref name="source"/> collection.
+		/// </summary>
+		/// <param name="source">The source collection to exclude elements from.</param>
+		/// <param name="startIndex">The zero-based index at which to begin excluding elements.</param>
+		/// <param name="count">The number of elements to exclude.</param>
+		internal ExcludeCollectionIterator(ICollection<T> source, int startIndex, int count)
+		{
+			_source = source;
+			_startIndex = startIndex;
+			_count = count;
+		}
+
+		/// <summary>
+		/// Creates a new collection iterator for excluding elements from the specified <paramref name="source"/> collection.
+		/// </summary>
+		/// <param name="source">The source collection to exclude elements from.</param>
+		/// <param name="range">The range of elements to exclude.</param>
+		internal ExcludeCollectionIterator(ICollection<T> source, Range range)
+		{
+			_source = source;
+			_startIndex = range.Start.Value;
+			_count = range.End.Value - _startIndex;
+		}
+
+		/// <summary>
+		/// Gets the number of elements in the source collection after excluding the specified portion of elements.
+		/// </summary>
+		public override int Count => _source.Count < _startIndex
+			? _source.Count
+			: _source.Count < _startIndex + _count
+				? _startIndex
+				: _source.Count - _count;
+
+		/// <inheritdoc cref="IEnumerable{T}" /> 
 		protected override IEnumerable<T> GetEnumerable() =>
-			ExcludeCore(source, startIndex, count);
+			ExcludeCore(_source, _startIndex, _count);
 	}
 
-	private sealed class ExcludeListIterator<T>(
-		IList<T> source,
-		int startIndex,
-		int count
-	) : ListIterator<T>
+	private sealed class ExcludeListIterator<T>
+		: ListIterator<T>
 	{
-		public override int Count =>
```

Comment:
When there's a chain of ternaries, I prefer this format everywhere, yes.

Reply:
Sounds good, I just noticed it was different from other ternaries in the file so I opted for consistency since I had to update it anyway. I'll change that format back 😄